### PR TITLE
WIP: Introduce operation concept and client pool

### PR DESCRIPTION
**Pull request by mbroadst**
This change set introduces a concept of an "operation" to our codebase, allowing us to abstract calls to libmongoc in order to do things like use a client pool, or asynchronously run the operation on a thread pool. I threw in the client pool to show how it can be used, maintaining existing behavior by reserving one client from the pool for things that haven't been converted to operations yet.

### CODE REVIEW

File: Sources/MongoSwift/APM.swift [hunk 1]
```diff
@@ -504,7 +504,7 @@ public enum MongoEventType {
     case serverMonitoring
 }
 
-/// An extension of MongoClient to add monitoring capability for commands and server discovery and monitoring.
+/// An extension of MongoClient to add monitoring capability for commands and server discovery and moni toring.
```
Comment (jmikola) [inline Sources/MongoSwift/APM.swift]: typo?

Comment (mbroadst) [reply Sources/MongoSwift/APM.swift]: yeah, this is a draft anyway to inform a design document

**Comment by kmahar** [comment]
is it possible to convert this to a draft, or can it not be reversed once it's "ready for review"?

**Comment by mbroadst** [comment]
I tried to convert it to a draft, but couldn’t find an easy way to do so.
I’ll just close the PR, you can look at the branch

**Pull request closed without merging**